src/test/java/plugins/MatrixAuthPluginTest.java

```diff
@@ -48,14 +49,23 @@ public void helloWorld() throws Exception {
         jenkins.login().doLogin("alice");
 
         FreeStyleJob j = jenkins.jobs.create();
+
         j.save();
 
         // if we login as Bob, he shouldn't see the job
         jenkins.login().doLogin("bob");
+
+        //wait for main panel to appear to make sure page is rendered
+        waitFor(by.id("main-panel"));
+
         assertNull(getElement(by.href("job/"+j.name+"/")));
 
         // contorl assertion: alice shoudl see the link
         jenkins.login().doLogin("alice");
+
+        //wait for main panel to appear to make sure page is rendered
+        waitFor(by.id("main-panel"));
```

Comment:
If you have a reference to a WebElement, and anything changes in this element, as soon as you try to use it, you'll get a StaleElementReferenceException. But this means you'll have to hold on to a reference of your webelement in your test. I suppose in the test harness this could be done since as far as I can see no Page Object Model is used.

You can then do `new WebDriverWait(driver, timeOut).until(ExpectedConditions.stalenessOf(element))` or whatever the test-harness equivalent is (if there is one).